core/backend/src/IModelDb.ts

```diff
@@ -1615,25 +1616,33 @@ export namespace IModelDb { // eslint-disable-line no-redeclare
       } finally {
         this._iModel.nativeDb.endPurgeOperation();
       }
-      // take another pass in case a deleted ViewDefinition was the only usage of these view-related DefinitionElements
-      const viewRelatedIds: Id64Array = [];
-      if (usageInfo.displayStyleIds) { viewRelatedIds.concat(usageInfo.displayStyleIds.filter((id) => usedIdSet?.has(id))); }
-      if (usageInfo.categorySelectorIds) { viewRelatedIds.concat(usageInfo.categorySelectorIds.filter((id) => usedIdSet?.has(id))); }
-      if (usageInfo.modelSelectorIds) { viewRelatedIds.concat(usageInfo.modelSelectorIds.filter((id) => usedIdSet?.has(id))); }
-      if (viewRelatedIds.length > 0) {
-        const viewRelatedUsageInfo = this._iModel.nativeDb.queryDefinitionElementUsage(viewRelatedIds);
-        if (viewRelatedUsageInfo) {
-          const usedViewRelatedIdSet: Id64Set | undefined = viewRelatedUsageInfo.usedIds ? Id64.toIdSet(viewRelatedUsageInfo.usedIds) : undefined;
-          try {
-            this._iModel.nativeDb.beginPurgeOperation();
-            deleteIfUnused(viewRelatedUsageInfo.displayStyleIds, usedViewRelatedIdSet);
-            deleteIfUnused(viewRelatedUsageInfo.categorySelectorIds, usedViewRelatedIdSet);
-            deleteIfUnused(viewRelatedUsageInfo.modelSelectorIds, usedViewRelatedIdSet);
-          } finally {
-            this._iModel.nativeDb.endPurgeOperation();
+      let notDeletedIds: Id64Set = new Set<Id64String>();
+      if (usageInfo.usedIds) {
+        notDeletedIds = Id64.toIdSet(usageInfo.usedIds);
+      }
+      if (usageInfo.viewDefinitionIds) {
+        // take another pass in case a deleted ViewDefinition was the only usage of these view-related DefinitionElements
+        let viewRelatedIds: Id64Array = [];
+        if (usageInfo.displayStyleIds) { viewRelatedIds = viewRelatedIds.concat(usageInfo.displayStyleIds.filter((id) => usedIdSet?.has(id))); }
```

Comment:
boring nitpick: formatting.

Reply:
Trying to keep this method from getting too long...

Reply:
Long readable function better than short less-readable function containing longer lines IMO but not gonna twist your arm.